When using a public github repository not registred in bower on heruko an exception is thrown

Bower supports a structure like this:

"dependencies": {
    "my-assets": "https://github.com/myorg/my-assets.git#master"
}
But trying this with my repo being public:

bower {
    'my-assets'('1.0.1') {
        url '"https://github.com/myrepo/my-assets.git'
    }
}
Gives me this exception:

Caused by: java.io.FileNotFoundException: https://bower.herokuapp.com/packages/my-assets
        at com.craigburke.gradle.client.registry.bower.BowerRegistry.loadInfoFromRegistry(BowerRegistry.groovy:88)
        at com.craigburke.gradle.client.registry.core.AbstractRegistry$_setInfo_closure4.doCall(AbstractRegistry.groovy:146)
        at com.craigburke.gradle.client.registry.core.AbstractRegistry$_setInfo_closure4.doCall(AbstractRegistry.groovy)
(This is running on the latest 1.3.0-SNAPSHOT build locally)

--- a/src/main/groovy/com/craigburke/gradle/client/registry/bower/BowerRegistry.groovy
+++ b/src/main/groovy/com/craigburke/gradle/client/registry/bower/BowerRegistry.groovy
@@ -85,8 +85,13 @@ class BowerRegistry extends AbstractRegistry implements Registry {
     @Override
     Map loadInfoFromRegistry(Dependency dependency) {
         URL url = new URL("${dependency.registry.url}/packages/${dependency.name}")
-        String jsonText = url.getText(requestProperties: ['User-Agent': DEFAULT_USER_AGENT])
-        new JsonSlurper().parseText(jsonText) as Map
+        try {
+            String jsonText = url.getText(requestProperties: ['User-Agent': DEFAULT_USER_AGENT])
+            return new JsonSlurper().parseText(jsonText) as Map
+        } catch (e) {
+            log.info("Dependency not found in $url ")
+            return null
+        }
     }
 
 }

--- a/src/main/groovy/com/craigburke/gradle/client/registry/core/AbstractRegistry.groovy
+++ b/src/main/groovy/com/craigburke/gradle/client/registry/core/AbstractRegistry.groovy
@@ -151,7 +151,7 @@ abstract class AbstractRegistry implements Registry {
                 infoFile.parentFile.mkdirs()
                 infoFile.text = new JsonBuilder(dependency.info).toPrettyString()
             }
-
+            dependency.info = dependency.info ?: [:]
             Resolver resolver = getResolver(dependency)
             resolver.afterInfoLoad(dependency)
         }

